### PR TITLE
lookup config by file path

### PR DESCRIPTION
**Pull request by gucong3000**
Fixed: #12
Fixed: #13
Fixed: #17

### CODE REVIEW

**Comment by gucong3000** [comment]
Fixed: #12
Fixed: #13

**Comment by gucong3000** [comment]
@Bubbi , @antsav , Can you help me do some test ?

unpack [zip file](https://github.com/gucong3000/atom-stylefmt/archive/patch-1.zip) to `~/.atom/packages/stylefmt/` and run

``` bash
cd ~/.atom/packages/stylefmt/
npm i --save stylefmt
```

**Comment by antsav** [comment]
followed instructions above

getting
`Refusing to install stylefmt as a dependency of itself`

and still not working

**Comment by antsav** [comment]
after all this manipulations , event 3.1.2 stopped working ))

**Comment by gucong3000** [comment]
`Refusing to install stylefmt as a dependency of itself`
@1000ch Same thing happend to me. can you rename to `atom-stylefmt` in `package.json` ?

**Comment by 1000ch** [comment]
@gucong3000 
Thank you for PR. Would you fix some points?
After fixing, merge your commits using `git rebase`.

> can you rename to atom-stylefmt in package.json ?

No, it should be `stylefmt`.

File: index.js [hunk 1]
```diff
@@ -48,16 +48,17 @@ function execute() {
...
   const position = editor.getCursorBufferPosition();
-  const paths = atom.project.getPaths();
+  const path = editor.getPath() || atom.project.getPaths()[0];
   const text = editor.getText();
-  const options = grammar === 'scss' ? {
-    syntax : scss
-  } : {};
+  const options = {
+    from: path
+  }
```
Comment (1000ch) [inline index.js:55]: Put semicolon and insert blank line after this.

File: index.js [hunk 2]
```diff
@@ -48,16 +48,17 @@ function execute() {
...
+  if(grammar === 'scss') {
+    options.syntax = scss
+  }
```
Comment (1000ch) [inline index.js:58]: Put semicolon.

File: index.js [hunk 3]
```diff
@@ -48,16 +48,17 @@ function execute() {
...
-  const paths = atom.project.getPaths();
+  const path = editor.getPath() || atom.project.getPaths()[0];
   const text = editor.getText();
-  const options = grammar === 'scss' ? {
-    syntax : scss
-  } : {};
+  const options = {
+    from: path
+  }
+  if(grammar === 'scss') {
```
Comment (1000ch) [inline index.js:56]: Space after `if`.

**Comment by gucong3000** [comment]
Closed & new PR for rebase #18

**Pull request closed without merging**